apps/party_management/test/pm_party_tests_SUITE.erl

```diff
@@ -1713,6 +1716,22 @@ compute_provider_terminal_terms_not_found(C) ->
             Client
         )).
 
+compute_provider_terminal_terms_undefined_terms(C) ->
+    Client = cfg(client, C),
+    DomainRevision = pm_domain:head(),
+    try
+        pm_client_party:compute_provider_terminal_terms(
+            ?prv(2),
+            ?trm(4),
+            DomainRevision,
+            #payproc_Varset{},
+            Client
+        )
+    catch
+        error:Error ->
+            ?assertMatch({{woody_error, {external, result_unexpected, _}}, _}, Error)
+    end.
```

Comment:
В таких случаях легче использовать [catch](https://erlang.org/doc/reference_manual/expressions.html#catch) завернутый в assert.

Reply:
Ты предлагаешь завернуть полностью `try .. catch`  в `?assertMatch`, или еще сократить 'try .. catch' до 'try Expr', как это сделано в соседних тестах?
Есть официальное мнение (книга Erlang Programming стр. 74), что конструкция (catch Expr) это легаси, и лучше использовать 'try .. catch'

> Using catch The original Erlang mechanism for exception handling was the catch. Because of its somewhat peculiar behavior, Richard Carlson, a member of the High Performance Er- lang Team at Uppsala University, suggested a review of Erlang’s exception handling, resulting in the try ... catch expression. He got the buy-in from the OTP team, which included this construct as a documented and permanent feature of the Erlang R10B release. We include a discussion of catch here because of the amount of legacy code that will use this construct. Be warned, however, that it’s not as elegant as try ... catch.